azavea/green-equity-demo
# Integrate real spending by month data

## Overview

Fetches and caches spending over time data for use by the animated map
Closes #35 

What's left:

- [x]  data integrity check ~~(ending aggregated data numbers match map values above, but significantly different from the BIL dashboard numbers. States are also not coloring as expected based on aggregated values at time so there might be some more tweaking between dummy data format and real format)~~
- [x] evaluate new color ranges that illustrate significant spending information

### Demo

https://user-images.githubusercontent.com/36374797/223159944-5304aa91-2dae-4c9a-a858-2291056a0816.mov

## Testing Instructions

- `scripts/update`
- Confirm `monthly.spending.json` file created and populated
- `scripts/server`
- Confirm animated map appears without loading and can play immediately
- Confirm map colors with new real spending data
    - no states currently hit above 10% of BIL spending, although CA will in a couple of months, so to test style updates to the dark blue color range locally we can update "now" to be a couple months in the future by adding `+ 3` to the [end of this line](https://github.com/azavea/green-equity-demo/blob/rm/add-fetch-data-by-time-script/src/app/src/components/AnimatedMap.tsx#L34).

 ## Checklist
